routing/segment.hpp

```diff
@@ -71,12 +79,108 @@ class Segment final
   bool m_forward = false;
 };
 
+class SegmentWeight final
+{
+public:
+  SegmentWeight() = default;
+
+  constexpr explicit SegmentWeight(double weight) : m_weight(weight) {}
```

Comment:
I'm not sure about explicit here too. It was useful while modification process cause if it's not explicit one can just left double where we really need SegmentWeight.